sds_in_a_box/SDSCode/opensearch_utils/client.py

```diff
@@ -130,12 +131,41 @@ def send_payload(self, payload):
         payload: Payload
             payload containing bulk documents to be sent to the OpenSearch cluster.
         """
-        self.client.bulk(payload.get_contents(), params={"request_timeout":1000000})
+        for chunk in payload.payload_chunks():
+            self.client.bulk(chunk, params={"request_timeout":1000000})
 
     def get_document(self, document):
         """Returns the specified document"""
         return self.client.get(index=document.get_index(), id=document.get_identifier())
 
+    def search(self, query, index):
+        """
+        Searches the OpenSearch cluster using the provided query object.
+
+        Parameters
+        ----------
+        query: Query
+            query object instantiated with the desired query parameters.
+        index: Index
+            OpenSearch index to use for the search.
+        """
+        # search the opensearch instance with scroll to handle larger responses
+        result = self.client.search(body=query.query_dsl(), index=index.get_name(), params={"scroll": "1m"})
+        scroll_id = result['_scroll_id']
+        scroll_size = len(result["hits"]["hits"])
+        counter = 0
+        full_result = result["hits"]["hits"]
+
+        # scroll through the results and add results to list
+        while scroll_size > 0:
```

Comment:
Sorry, responded to the other comment before I saw this. Correct, there's a request limit that can get hit when querying for large amounts of data. The scroll features helps to avoid that limit.